piet/src/text.rs

```diff
@@ -42,17 +45,46 @@ pub trait TextLayout {
     /// Measure the advance width of the text.
     fn width(&self) -> f64;
 
-    /// Given a Point, determine the corresponding text position
+    /// Given a `Point`, determine the corresponding text position.
+    ///
+    /// ## Return value:
+    /// Returns a `HitTestPoint` describing the results of the test.
```

Comment:
It's actually probably worth making these links to the structs in question; I saw you doing this below for some other things and it's helpful.

I also find that for the purposes of reading the docs in the code itself (as opposed to the rendered HTML docs) it can be nice to put all the links down at the bottom of the comment, like:

```rust
/// link to [`HitTestPoint`]
/// 
/// at the bottom of the whole comment, after a newline, add:
///
/// [`HitTestPoint`]: struct.HitTestPoint.html
```

If we do that, maybe we can just document the fields on the struct? That might be simpler? It would save us having to document these twice for these two functions, at least.

Reply:
Thanks for the example, that really cleans things up.

I'm going to add the links, but for now I'm going to keep the field documentation in both the method and the return struct. My reasoning is that it's a little harder to digest the docs if you have to flip back and forth between pages.

Open to changing in the future.